packages/hint-create-element-svg/tests/tests.ts

```diff
@@ -0,0 +1,49 @@
+import generateHTMLPage from 'hint/dist/src/lib/utils/misc/generate-html-page';
+import { getHintPath } from 'hint/dist/src/lib/utils/hint-helpers';
+import { HintTest } from '@hint/utils-tests-helpers/dist/src/hint-test-type';
+import * as hintRunner from '@hint/utils-tests-helpers/dist/src/hint-runner';
+
+const hintPath = getHintPath(__filename);
+
+const generateHTMLPageWithDivTag = (head: string, body: string) => {
+    return generateHTMLPage(head, `<div id="container">${body}</div>`);
+};
+const generateScriptTag = (script: string) => {
+    return `<script type="text/javascript">${script}</script>`;
+};
+
+const htmlElementCreate = 'document.getElementById("container").appendChild(document.createElement("a"));';
+const invalidCircleCreate = `const container = document.getElementById("container")
+                             const circle = document.createElement("circle");
+                             container.appendChild(circle);`;
+const invalidSvgCreate = 'document.getElementById("container").appendChild(document.createElement("svg"));';
+const validSvgCreate = 'document.getElementById("container").appendChild(document.createElementNS("http://www.w3.org/2000/svg", "svg"));';
+
+const tests: HintTest[] = [
+    {
+        name: 'Can create HTML element using createElement',
+        serverConfig: generateHTMLPageWithDivTag('', generateScriptTag(htmlElementCreate))
+    },
+    {
+        name: 'Should not use createElement to create SVG element',
+        reports: [{
+            message: 'SVG elements cannot be created with createElement; use createElementNS instead',
+            position: { match: 'createElement("svg")' }
+        }],
+        serverConfig: generateHTMLPageWithDivTag('', generateScriptTag(invalidSvgCreate))
+    },
+    {
+        name: 'Can use createElementNS to create SVG element',
+        serverConfig: generateHTMLPageWithDivTag('', generateScriptTag(validSvgCreate))
+    },
+    {
+        name: 'Should not use createElement to create Circle SVG element',
+        reports: [{
+            message: 'SVG elements cannot be created with createElement; use createElementNS instead',
+            position: { match: 'createElement("circle")' }
+        }],
+        serverConfig: generateHTMLPageWithDivTag('', generateScriptTag(invalidCircleCreate))
+    }
+];
+
+hintRunner.testHint(hintPath, tests, { parsers: ['javascript']});
```

Comment:
Probably also good to add one test validating this works as expected with external script files. You can check out the tests in `hint-css-prefix-order` for an example of how to set up a `serverConfig` with external files.